### ISSUE TITLE
Missing struct auto-complete on definition

### ISSUE THREAD

**Issue by nunojsa** [microsoft/vscode-cpptools#4610]
- OS: Arch Linux and Ubuntu18.04
- VS Code Version: 1.40
- C/C++ Extension Version: 0.26.2
- Other extensions you installed (and if the issue persists after disabling them): Docker, Markdown, CMake, DeviceTree (and the issue persists after disabling them).

I guess this is more a feature request than actually an issue.
This feature is something I had in previous IDE's and, although it's not the most critical thing, I find it pretty useful. So, in C (not sure if this is allowed in c++ - it should be fine I guess), you can do the following:

```
struct foo {
       int a;
       int b;
};

static struct foo temp = {
       .a = 10, (don't get the member list when typing .)
       .b = 20,
};
```

In this situation, the member's autocomplete is not working. Autocomplete here is pretty useful since you can have a look of all the members and decide which ones you want to initialize. This is a pretty normal way of defining things in the linux kernel for example. 
Moreover, if the error squiggles are enabled, it will complain even though it's perfectly legal to do it in c. I believe it will throw a warning in c90 but, hey, that's old :smile: 

It would be nice to have this.
(Not sure if this is supported already and it's just not working for me...)

Thanks!

**Comment by sean-mcmanus** [comment]
I filed a bug for VS at https://developercommunity.visualstudio.com/content/problem/822907/cc-intellisense-list-members-doesnt-work-with-desi.html .

We added support for designated initializers for C++20 (see https://github.com/Microsoft/vscode-cpptools/issues/3491 ). However, we may need to more support for the C99 extension.

**Comment by Colengms** [comment]
This appears to have been fixed (in 0.29.0).  Closing as fixed.
